tests/test_application.py

```diff
@@ -1680,3 +1680,37 @@ async def raise_method(*args, **kwargs):
                 app.run_webhook(close_loop=False, stop_signals=None)
 
         assert len(recwarn) == 0
+
+    def test_signal_handlers(self, app, monkeypatch):
+        # this test should make sure that signal handlers are set by default on Linux + Mac,
+        # and not on Windows.
+
+        def signal_handler_test(*args, **kwargs):
+            # args[0] is the signal, [1] the callback
+            assert args[0] in (signal.SIGINT, signal.SIGTERM, signal.SIGABRT)
+
+        loop = asyncio.get_event_loop()
+        monkeypatch.setattr(loop, "add_signal_handler", signal_handler_test)
+
+        async def abort_app(context):
+            raise KeyboardInterrupt
+
+        app.job_queue.run_once(abort_app, 2)
+        app.run_polling(close_loop=False)
+
+        app.job_queue.run_once(abort_app, 2)
+        app.run_webhook(port=49152, webhook_url="example.com", close_loop=False)
+
+        if platform.system() == "Windows":
+
+            def signal_handler_test_windows(*args, **kwargs):
+                # this should be empty, but still called once.
+                assert args == []
+
+            monkeypatch.setattr(loop, "add_signal_handler", signal_handler_test_windows)
+            # this should run through, and we expect the signals to be empty
+            app.job_queue.run_once(abort_app, 2)
+            # pass false so it is not default
+            app.run_webhook(
+                port=49152, webhook_url="example.com", close_loop=False, stop_signals=False
+            )
```

Comment:
sorry, but this still doesn't do it IMO. `stop_signals=False` doesn't make sense because we do in fact want to test the default case. Moreover, `add_signal_handler` is never called (as intended), so the `assert` in the mocked function doesn't test anything. Granted, it raises an AssertionError if a code change accidently passes a signal on windows (only without `stop_signal=False` though), but that's really hard to read from this test. Additionally, it feels unncessary to have the above tests do nothing on windows (which again is not obvious from the tests) and only this if-clause actually test something.

What speaks against implementing the test as suggested in https://github.com/python-telegram-bot/python-telegram-bot/pull/3065/files#r879864569 ?

Reply:
> What speaks against implementing the test as suggested 

me forgetting you suggested that.